test/integration-serial/resource_discovery_test.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class ResourceDiscoveryTest < KubernetesDeploy::IntegrationTest
+  def test_non_prunable_crd
+    assert_deploy_success(deploy_fixtures("resource-discovery", subset: ["shredders.yml"]))
+    assert_deploy_success(deploy_fixtures("resource-discovery", subset: ["shredders_cr.yml"]))
+    # Deploy any other non-priority (predeployable) resource to trigger pruning
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: ["configmap-data.yml"]))
+
+    refute_logs_match("The following resources were pruned: widget \"my-first-shredder\"")
+  end
```

Comment:
don't forget to clean up the CRDs. I'm thinking we should make test teardown do this generically by listing and deleting CRDs. That could just be in this suite if we move the other CRD tests over.